bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java

```diff
@@ -1243,73 +1286,223 @@ public void sendNotificationToMobileApp(String customerId, String text, @Nullabl
+                target.devices = targetDevices;
+                parameters.put("target", target);
 
-        String accountCustomerId = this.accountCustomerId;
-        String customerId = StringUtils.isEmpty(accountCustomerId) ? device.deviceOwnerCustomerId : accountCustomerId;
+                String accountCustomerId = this.accountCustomerId;
+                String customerId = StringUtils.isEmpty(accountCustomerId)
+                        ? devices.toArray(new Device[devices.size()])[0].deviceOwnerCustomerId
+                        : accountCustomerId;
 
-        if (customerId != null) {
-            parameters.put("customerId", customerId);
+                if (customerId != null) {
+                    parameters.put("customerId", customerId);
+                }
+                executeSequenceCommandWithVolume(devices.toArray(new Device[devices.size()]), "AlexaAnnouncement",
+                        parameters, ttsVolumes.toArray(new Integer[ttsVolumes.size()]),
+                        standardVolumes.toArray(new Integer[standardVolumes.size()]));
+            } catch (IOException | URISyntaxException e) {
+                logger.error("send textToSpeech fails with unexpected error", e);
+            } finally {
+                announcementSenderUnblockFuture = scheduler.schedule(this::queuedAnnouncement, speak.length() * 100,
+                        TimeUnit.MILLISECONDS);
+            }
+        } else {
+            announcementQueueRunning.set(false);
+            announcementSenderUnblockFuture = null;
         }
-        executeSequenceCommandWithVolume(device, "AlexaAnnouncement", parameters, ttsVolume, standardVolume);
     }
 
-    public void textToSpeech(Device device, String text, @Nullable Integer ttsVolume, int standardVolume)
-            throws IOException, URISyntaxException {
-        Map<String, Object> parameters = new HashMap<>();
-        parameters.put("textToSpeak", text);
-        executeSequenceCommandWithVolume(device, "Alexa.Speak", parameters, ttsVolume, standardVolume);
+    public synchronized void textToSpeech(Device device, String text, @Nullable Integer ttsVolume,
+            @Nullable Integer standardVolume) {
+        if (text == null || text.replaceAll("<.+?>", "").trim().isEmpty() || ttsVolume == null || standardVolume == null) {
+            return;
+        }
+        if (textToSpeechTimer != null) {
+            textToSpeechTimer.cancel(true);
+        }
+        JsonTextToSpeech jsonTextToSpeech = new JsonTextToSpeech();
+        jsonTextToSpeech.text = text;
+        String json = gson.toJson(jsonTextToSpeech);
+        if (!textToSpeeches.containsKey(json)) {
+            jsonTextToSpeech.devices = new ArrayList<>();
+            jsonTextToSpeech.ttsVolumes = new ArrayList<>();
+            jsonTextToSpeech.standardVolumes = new ArrayList<>();
+            textToSpeeches.put(json, jsonTextToSpeech);
+        } else {
+            jsonTextToSpeech = textToSpeeches.get(json);
+        }
```

Comment:
```suggestion
        jsonTextToSpeech = textToSpeeches.computeIfAbsent(json, k -> { 
            jsonTextToSpeech.devices = new ArrayList<>();
            jsonTextToSpeech.ttsVolumes = new ArrayList<>();
            jsonTextToSpeech.standardVolumes = new ArrayList<>();
        });
```

Reply:
It seems this doesn't work, got a compilation error. Never used this function before.

Reply:
Yeah. Didn't think about effective final. I have mad a suggestions via PR to your repo.